apps/homescreen/index.html

```diff
@@ -33,39 +37,59 @@
 
     <link rel="stylesheet" href="style/window.css">
     <link rel="stylesheet" href="style/grid.css">
+    <link rel="stylesheet" href="style/apps-panel.css">
+    <link rel="stylesheet" href="style/pages-panel.css">
+
+    <!-- Process the icons of pinned pages -->
+    <script defer src="shared/elements/gaia_grid/js/grid_icon_renderer.js"></script>
+    <script defer src="shared/js/homescreens/icon.js"></script>
+    <script defer src="shared/js/icons_helper.js"></script>
 
     <script defer src="js/metadata.js"></script>
     <script defer src="js/datastore.js"></script>
+    <script defer src="js/pagesstore.js"></script>
+    <script defer src="js/pages.js"></script>
     <script defer src="js/app.js"></script>
     <script defer src="js/bootstrap.js"></script>
   </head>
 
   <body>
-    <div id="shadow"></div>
-    <div id="scrollable">
-      <gaia-container id="apps" drag-and-drop></gaia-container>
-    </div>
-    <div id="bottombar">
-      <div id="uninstall"></div>
-      <div id="edit"></div>
-    </div>
-    <gaia-dialog id="cancel-download" class="dialog-action">
-      <section>
-        <p class="body" data-l10n-id="stop-download-body"></p>
-      </section>
-      <button class="action" data-l10n-id="stop-download-action"></button>
-      <button on-click="close" data-l10n-id="cancel-action"></button>
-    </gaia-dialog>
-    <gaia-dialog id="resume-download" class="dialog-action">
-      <section>
-        <p class="body" data-l10n-id="resume-download-body"></p>
-      </section>
-      <button class="action" data-l10n-id="resume-download-action"></button>
-      <button on-click="close" data-l10n-id="cancel-action"></button>
-    </gaia-dialog>
-    <gaia-dialog id="settings" class="dialog-action">
-      <button class="action" data-l10n-id="homescreen-settings"></button>
-      <button on-click="close" data-l10n-id="cancel-action"></button>
-    </gaia-dialog>
+  <div id="page-indicator"><span class="active"></span><span></span></div>
+    <div id="panels"><!--
```

Comment:
Yes, the white-space ends up inserting actual space in the page layout - I could work around this in CSS, but rather than complicating that, it's easier to just remove the white-space in the document with comments.